--- /usr/local/emhttp/plugins/dynamix.vm.manager/include/libvirt_helpers.php.original
+++ /usr/local/emhttp/plugins/dynamix.vm.manager/include/libvirt_helpers.php
@@ -431,6 +431,11 @@
		'Ubuntu' => [
			'form' => 'Custom.form.php',
			'icon' => 'ubuntu.png',
			'os' => 'ubuntu'
		],
+
+		'UnRAID' => [
+			'form' => 'UnRAID.form.php',
+			'icon' => 'unraid.png'
+		],


 		' Linux ' => '', /* Linux Header */
@@ -428,6 +433,10 @@
 			'icon' => 'ubuntu.png',
 			'os' => 'ubuntu'
 		],
+		'UnRAID' => [
+			'form' => 'UnRAID.form.php',
+			'icon' => 'unraid.png'
+		],
 		'VyOS' => [
 			'form' => 'Custom.form.php',
 			'icon' => 'vyos.png',
@@ -441,6 +450,27 @@
 			'icon' => 'default.png'
 		]
 	];
+
+	$arrUnRAIDVersions = json_decode(file_get_contents('https://releases.unraid.net/usb-creator'), true);
+	$arrUnRAIDVersions = array_merge(...array_map(
+		fn($os) => $os['subitems'] ?? [$os],
+		$arrUnRAIDVersions['os_list']
+	));
+
+	$arrUnRAIDVersions = array_column(
+		array_map(
+			fn($item) => [
+				'name' => substr($item['name'], 7),
+				'url' => $item['url'],
+				'size' => $item['image_download_size'],
+				'localpath' => '',
+				'valid' => 0
+			],
+			array_filter($arrUnRAIDVersions, fn($item) => $item['name'] !== 'Unraid OS Releases (Next Branch)')
+		),
+		null,
+		'name'
+	);


 	$fedora = '/var/tmp/fedora-virtio-isos';
